ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessRepository.scala

```diff
@@ -78,22 +81,31 @@ class DBProcessRepository(val dbConfig: DbConfig, val modelVersion: ProcessingTy
         case Some(_) => DBIOAction.successful(ProcessAlreadyExists(action.processName.value).asLeft)
         case None => processesTable.filter(_.name === action.processName.value).result.headOption.flatMap {
           case Some(_) => DBIOAction.successful(ProcessAlreadyExists(action.processName.value).asLeft)
-          case None => (insertNew += processToSave).flatMap(entity => updateProcessInternal(ProcessId(entity.id), action.processDeploymentData))
+          case None => (insertNew += processToSave)
+            .flatMap(entity => updateProcessInternal(ProcessId(entity.id), action.processDeploymentData))
+            .map(_.right.map(_.newVersion))
         }
       }
     }
 
     insertAction
   }
 
-  def updateProcess(updateProcessAction: UpdateProcessAction)(implicit loggedUser: LoggedUser): DB[XError[Option[ProcessVersionEntityData]]] =
+  def updateProcess(updateProcessAction: UpdateProcessAction)(implicit loggedUser: LoggedUser): DB[XError[ProcessUpdated]] = {
```

Comment:
I wonder about move adding comment in transaction way at ProcessService. WDYT?

Reply:
I added an comment about extracting logic.